rotkehlchen/tests/unit/test_savepoints.py

```diff
@@ -0,0 +1,44 @@
+import sqlite3
+
+import pytest
+
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+
+
+def test_savepoints():
+    conn = GlobalDBHandler().conn
+    conn.execute('CREATE TABLE a(b INTEGER PRIMARY KEY)')
+    with pytest.raises(sqlite3.Error):
+        with conn.savepoint_ctx('point1') as cursor1:
+            assert conn.savepoints == ['point1']
+            cursor1.execute('INSERT INTO a VALUES (1), (2)')  # insert something
+            with conn.savepoint_ctx('point2') as cursor2:
+                assert conn.savepoints == ['point1', 'point2']
+                # insert more and check multiple nested savepoints
+                cursor2.execute('INSERT INTO a VALUES (3)')
+                with pytest.raises(sqlite3.Error):
+                    # Also check savepoints with the same name. In this case, according to sqlite's
```

Comment:
We can and should protect against this as I said in the other part of the review